server/api/controllers/question.js

```diff
@@ -0,0 +1,22 @@
+const Question = require("../models/question");
+
+exports.get = (req, res) => {
+  const limit = Number(req.query.limit);
+
+  if (!limit) {
+    res.status(404).json({
+      error: { message: "limit is not provided" }
+    });
+  }
+
+  let options = { limit: limit };
+  Question.findRandom({}, {}, options, (err, quotes) => {
+    if (!err) {
+      res.send(quotes);
+    } else {
+      res.status(500).json({
```

Comment:
Let's reverse the check and return the exact error if there is error.

Later if there are no quotes return 200 status with empty quotes array.

And finally return the quotes.

Reply:
@skyshader I don't think its a good idea to return the exact server error to the client. Instead we can  throw the error on the server machine and send something like "Server error occurred" with a 500